### PR TITLE
My Home: Refactor Grow and Earn into its own component.

### PR DESCRIPTION
**Pull request by kwight**
This PR moves the Grow & Earn card to its own component, in preparation for cards being displayed according to server-side logic. This continues the work in #40368 . No visual or behavioural changes should occur.

Screen Shot 2020-03-25 at 4 41 07 PM

**Testing Instructions**

*  Load this branch, and go to `/home/:site` for a site that is launched and has a complete checklist.
* Verify it visually looks the same in production (load the same route on `wordpress.com`).
* Check the card links, and verify that stats fire.

### CODE REVIEW

File: client/my-sites/customer-home/cards/features/grow-earn/index.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+				)
+			),
+		expandToolsSection: () => dispatch( expandSection( SIDEBAR_SECTION_TOOLS ) ),
+	} ),
+	( stateProps, dispatchProps, ownProps ) => ( {
+		...stateProps,
+		...ownProps,
+		expandToolsAndTrack: ( section, action ) => {
+			dispatchProps.expandToolsSection();
+			dispatchProps.trackAction( section, action, stateProps.isStaticHomePage );
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/features/grow-earn/index.jsx]: Whoops I got the ordering wrong in other spots. The default method is
https://github.com/reduxjs/react-redux/blob/172b2334d3a1240248c049a40b88a845bd0ecf5d/src/connect/mergeProps.js#L4

File: client/my-sites/customer-home/cards/features/grow-earn/style.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.grow-earn {
+    padding-bottom: 12px;
+
+	&__card-subheader {
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/features/grow-earn/style.scss]: Let's avoid parent selectors here too 🙂 
```suggestion
	.grow-earn__card-subheader {
```

File: client/my-sites/customer-home/cards/features/grow-earn/index.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ */
+import './style.scss';
+
+export const GrowEarn = ( { siteSlug, expandToolsAndTrack } ) => {
+	const translate = useTranslate();
+
+	return (
+		<Card className="grow-earn">
+			<CardHeading>{ translate( 'Grow & Earn' ) }</CardHeading>
+			<h6 className="grow-earn__card-subheader">
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/features/grow-earn/index.jsx]: We could also keep using `customer-home__card-subheader` here so we don't need to copy shared styles across all components.

To get rid of the linter issue in that case, we can use both classes: `grow-earn__card-subheader customer-home__card-subheader`.

Comment (kwight) [reply client/my-sites/customer-home/cards/features/grow-earn/index.jsx]: > To get rid of the linter issue in that case, we can use both classes: grow-earn__card-subheader customer-home__card-subheader.

Ah, I didn't even think of that! I admit to quietly bristling every time we change code to something less optimal/accurate for the sake of a digital overlord, but that's a reasonable-ish solution (still silly, but better than duping styles). 👍

Comment (kwight) [reply client/my-sites/customer-home/cards/features/grow-earn/index.jsx]: 9d52b0e28d

File: client/my-sites/customer-home/cards/features/grow-earn/style.scss (hunk 2 above)
Comment (kwight) [reply client/my-sites/customer-home/cards/features/grow-earn/style.scss]: 9d52b0e28d

File: client/my-sites/customer-home/cards/features/grow-earn/index.jsx (hunk 1 above)
Comment (kwight) [reply client/my-sites/customer-home/cards/features/grow-earn/index.jsx]: e1ba3e2eff

**Pull request merged**